### ISSUE TITLE
There should be an option to skip COM812 (missing-trailing-comma) on single-item calls/lists. [Feature request]

### ISSUE THREAD

**Issue by echo-lalia** [astral-sh/ruff#13410]
Currently, the rule `COM812` will flag any calls/lists where the final item has no comma.  
For the most part, this is useful and I like the consistency it provides. However, there are two issues with this rule that keep popping up in my code:  

Calls with only one item, that can *never* have another item added, are still flagged. This is a matter of preference, but I find this to actually harm readability, because it suggests that this is a list of items, where other items may be provided afterwards, and that is often not the case. Example:
``` Py
float(
    width
    * height
    / step,  # <- I find this comma more confusing/distracting than helpful
)
```

My second issue with this is MicroPython-specific *(which I realize may be out-of-scope for this project, but it causes problems for me so I think it's worth mentioning)*. This rule also recommends adding commas to constants, which breaks the code until the offending comma is removed. Example:
``` Py
_DISPLAY_PADDING = const(
    (_DISPLAY_HEIGHT - (6 * _SMALL_FONT_HEIGHT + _NUM_LINES * _FONT_HEIGHT))
    // (_NUM_LINES),
)
# SyntaxError: not a constant
```

<br/>

If there was an option to disable the comma rule when only a single value is present, then both of these issues would be solved.  

*Alternatively: if there was an optional list of names to ignore this rule inside of, then both `float` and `const` could be ignored, and both of these issues would be solved.*

**Comment by MichaReiser** [comment]
Hi @echo-lalia 

Hmm, I can see how this is problematic but I think these two cases are exactly what the rule aims to catch

> Calls with only one item, that can never have another item added, are still flagged.

I can see how this might be desired, but it requires very involved analysis for a rule to determine if a callee has only a single parameter. Resolving the callee's signature would make the rule significantly slower. Resolving the callee will only work reliably for typed projects. 

But this also ignores the fact that code evolves. A function that only accepts a single parameter may accept multiple parameters tomorrow. Having the trailing comma in place reduces the diff where the new arguments are added because the formatting of the first argument remains unchanged. 

> My second issue with this is MicroPython-specific (which I realize may be out-of-scope for this project, but it causes problems for me so I think it's worth mentioning). This rule also recommends adding commas to constants, which breaks the code until the offending comma is removed. Example:

This certainly is frustrating. I don't know MicroPython myself and what there goal is but have you considered to open an issue and asking them to make `const` function calls python spec compliant so that they accept optional trailing commas?

**Comment by echo-lalia** [comment]
Thanks for taking the time to respond and to explain!

I have no understanding of how Ruff works internally, and so when I submitted this issue, I had assumed this might be a relatively simple addition.  
Clearly that wasn't quite right, so I appreciate you explaining as much :)

<br/>

> A function that only accepts a single parameter may accept multiple parameters tomorrow. Having the trailing comma in place reduces the diff where the new arguments are added...

It seems like my suggestion probably won't be added, but I realized I forgot another funny example, and so I wanted to add it on now for the sake of discussion:
``` Py
handle_text(
    "This is a writing style that has come up in my projects on occasion.\n"
    "It allows indentation to be maintained with surrounding code, "
    "And for lines to be formatted in the code differently than in the result.\n"
    f"{' '.join(('You', 'can', 'also', 'use', 'it', 'to', 'split', 'up', 'f-strings'))}.\n"
    "And if you add a comma to the end, it makes the diff larger when you change the text.",
    )
```

Obviously this isn't the end of the world, but I thought it was worth adding to the post.

<br/>

> have you considered to open an issue and asking them to make const function calls python spec compliant so that they accept optional trailing commas?

I think it would definitely be sensible for it to be fixed on the MicroPython side :) It's a rapidly evolving project, so it's not unlikely that this gets fixed at some point. `const` variables are actually 'real' constants that get evaluated at compile time, and so I imagined there could be a really important (but unknown to me) reason that it doesn't understand the comma. But, I don't actually know because I haven't asked!

**Comment by MichaReiser** [comment]
That's a fair point. It's a trade-off between optimizing the diff size for adding a new argument or appending to the string. In this case, it's not possible to have both. 

A formatting that avoids this is to assign the message to a variable where `COM812` doesn't apply.

```python
msg = (
    "This is a writing style that has come up in my projects on occasion.\n"
    "It allows indentation to be maintained with surrounding code, "
    "And for lines to be formatted in the code differently than in the result.\n"
    f"{' '.join(('You', 'can', 'also', 'use', 'it', 'to', 'split', 'up', 'f-strings'))}.\n"
    "And if you add a comma to the end, it makes the diff larger when you change the text."
)

handle_text(msg)
```

Or you could use a library like `textwrap` to remove the indent at runtime (with the downside that it comes with a performance penalty).

```python
handle_text(
    textwrap.dedent(f"""
        This is a writing style that has come up in my projects on occasion.
        It allows indentation to be maintained with surrounding code,
        And for lines to be formatted in the code differently than in the result.
        {' '.join(('You', 'can', 'also', 'use', 'it', 'to', 'split', 'up', 'f-strings'))}.
        And if you add a comma to the end, it makes the diff larger when you change the text.
    """
    )
)
```
